src/portal/src/app/config/gc/gc.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable, Subscription, Subject, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { GcApiRepository } from './gc.api.repository';
+import { ErrorHandler } from '@harbor/ui';
+import { GcJobData } from './gcLog';
+
+@Injectable()
+export class GcRepoService {
+
+    constructor(private http: Http,
+        private gcApiRepository: GcApiRepository,
+        private errorHandler: ErrorHandler) {
+    }
+
+    public manualGc(): Observable <any> {
+        let param = {
+            "schedule": {
+                "type": "Manual"
+            }
+        };
+        return this.gcApiRepository.postSchedule(param);
```

Comment:
repository are directly interact with backend apis. For the service, there are some business level logic inside it. It can be more clear to do like this. Many applications use this 3 layers in frontend .